### PR TITLE
Test shm usage extension for transient local qos

### PR DESCRIPTION
**Pull request by MatthiasKillat**

### CODE REVIEW

**Comment by MatthiasKillat** [comment]
Requires https://github.com/eclipse-cyclonedds/cyclonedds/pull/1180/files

**Comment by MatthiasKillat** [comment]
Test cases to check the scenarios for shared memory usage which are enabled by https://github.com/eclipse-cyclonedds/cyclonedds/pull/1180/

File: src/ddscxx/tests/SharedMemory.cpp [hunk 1]
```diff
@@ -438,21 +448,161 @@ TYPED_TEST(SharedMemoryTest, invalid_shm_qos)
...
+TYPED_TEST(SharedMemoryTest, best_effort_reader_receives_data_from_reliable_writer_via_shm)
+{
+  dds::sub::qos::DataReaderQos r_qos;
+  r_qos << dds::core::policy::Reliability::BestEffort();
+  r_qos << dds::core::policy::Durability::TransientLocal();
+  r_qos << dds::core::policy::History::KeepLast(10U);
+  constexpr bool valid_r_shm_qos = true;
+
+  // to set the durability service values of history depth
+  // TODO: except that this does not work, as the writer has no
```
Comment (MatthiasKillat) [inline src/ddscxx/tests/SharedMemory.cpp:529]: @sumanth-nirmal Either I use it wrong or it does not work (it compiles, but the value is not propagated as expected). This makes testing of https://github.com/eclipse-cyclonedds/cyclonedds/pull/1180/ harder.

Comment (MatthiasKillat) [reply src/ddscxx/tests/SharedMemory.cpp:529]: Needs to be understood/fixed before we consider merging this.

Comment (MatthiasKillat) [reply src/ddscxx/tests/SharedMemory.cpp]: Fixed and are now using DurabilityService QoS

**Comment by MatthiasKillat** [comment]
@eboasson Should now correctly use the durability service as intended. Tests pass locally with https://github.com/eclipse-cyclonedds/cyclonedds/pull/1180.

**Review by eboasson — APPROVED** [review]
As expected, CI turned green after merging eclipse-cyclonedds/cyclonedds#1180 🙂 Thanks @MatthiasKillat !

**Pull request merged**